frontend/src/components/elements/Text/Text.tsx

```diff
@@ -89,30 +93,43 @@ class Text extends React.PureComponent<Props> {
       link: linkWithTargetBlank,
       linkReference: linkReferenceHasParens,
     }
+    const styleProp = { width }
 
     switch (format) {
       // Plain, fixed width text.
-      case TextProto.Format.PLAIN:
-        return (
-          <div className="fixed-width stText" style={{ width }}>
-            {body}
-          </div>
-        )
+      case TextProto.Format.PLAIN: {
+        const allowHTML = element.get("allowHtml")
+
+        const props = {
+          className: classNames("fixed-width", "stText"),
+          style: styleProp,
+        }
+
+        if (allowHTML) {
+          return (
+            <div
+              {...props}
+              dangerouslySetInnerHTML={{
```

Comment:
Let's not do this. This is a big security hole.

Our goal with `allow_unsafe_html` wasn't to provide a way for users to do absolutely anything they wanted. Instead we were actually preparing to deprecate an existing feature (Markdown HTML support) due to security concerns. But if we just removed that feature we would break some users' apps, and that's no good. So a temporary solution was to give people an escape hatch while we developed more secure solutions for them. Let's not move back to a less model now.

Regarding bug #247, the Markdown component we are using should already be able to support `<span style="color: blue">`. You can try it out [here](https://rexxars.github.io/react-markdown/). Their HTML support is not perfect, but it provides a little bit of filtering against obviously bad HTML code, like `<script>` tags.

You can do  `<span style="color: blue">` in the link above but not in Streamlit, so there seems to be something funny about how we're using that component. Let's fix _that_ instead.

Reply:
Resolved using htmlParser plugin, but there is an existing module in which you can specify allowed tags  to be rendered , that could be a possibilty https://www.npmjs.com/package/sanitize-html instead of using markdown.